lib/kube/proxy/auth.go

```diff
@@ -174,6 +174,19 @@ func extractKubeCreds(ctx context.Context, cluster string, clientCfg *rest.Confi
 	if err != nil {
 		return nil, trace.Wrap(err, "failed to generate TLS config from kubeconfig: %v", err)
 	}
+	if tlsConfig == nil {
+		cc := rest.CopyConfig(clientCfg)
```

Comment:
Do we really need to echo the entire config? It seems a bit dangerous to do.

Maybe use rest.AnonymousClientConfig? Although I'd rather not echo everything if possible.